eval.go

```diff
@@ -220,6 +230,14 @@ nextFileMatch:
 
 		md := d.repoMetaData[d.repos[nextDoc]]
 
+		if d.repos[nextDoc] != curRepo {
```

Comment:
Could we make this comparison slightly cheaper by comparing repo IDs?

```go
if md.RepositoryID != repoID { ... }
```

Reply:
md.ID is a uint32 and d.repos[nextDoc] is uint16. Do you mean it is cheaper because of the slice lookup?

Reply:
Never mind, I got tricked by search based code intelligence telling me it was a simpleSection. But yeah, you could save one slice look up.